src/Adapters/DatabaseAdapter.php

```diff
@@ -145,4 +146,51 @@ public function removeFilteredPolicy(string $sec, string $ptype, int $fieldIndex
 
         $instance->destroy();
     }
+
+    /**
+     * adds a policy rules to the storage.
+     * This is part of the Auto-Save feature.
+     *
+     * @param string $sec
+     * @param string $ptype
+     * @param array  $rules
+     */
+    public function addPolicies(string $sec, string $ptype, array $rules): void
+    {
+        $cols = [];
+
+        foreach($rules as $rule) {
+            $temp = [];
+            $temp['ptype'] = $ptype;
+            foreach ($rule as $key => $value) {
+                $temp['v'.strval($key)] = $value;
+            }
+            $cols[] = $temp;
+        }
+
+        RulesModel::create()->saveAll($cols);
+    }
+
+    /**
+     * removes policy rules from the storage.
+     * This is part of the Auto-Save feature.
+     *
+     * @param string $sec
+     * @param string $ptype
+     * @param array  $rules
+     */
+    public function removePolicies(string $sec, string $ptype, array $rules): void
+    {
+        $ids = [];
+        foreach($rules as $rule) {
+            $where = [];
+            $where['ptype'] = $ptype;
+            foreach ($rule as $key => $value) {
+                $where['v'.strval($key)] = $value;
+            }
+            $ids[] = RulesModel::create()->get($where)->id;
```

Comment:
Change it to the following code:
```php
$ret = RulesModel::create()->get($where);
$ret && $ids[] = $ret->id;
```

Reply:
@basakest is it resolved?

Reply:
@hsluoyz yes

Reply:
Then click 
![image](https://user-images.githubusercontent.com/3787410/113285244-4ff93980-931d-11eb-899d-47b55c3c1ba0.png)